core/server/worker/src/main/java/alluxio/worker/block/UnderFileSystemBlockStore.java

```diff
@@ -266,89 +261,52 @@ public BlockReader createBlockReader(final long sessionId, long blockId, long of
   }
 
   /**
-   * @param sessionId the session ID
-   * @param blockId the block ID
-   * @return true if mNoCache is set
+   * A token for a block that has been granted access via
+   * {@link #acquireAccess(long, long, Protocol.OpenUfsBlockOptions)}.
    */
-  public boolean isNoCache(long sessionId, long blockId) {
-    Key key = new Key(sessionId, blockId);
-    BlockInfo blockInfo = mBlocks.get(key);
-    UnderFileSystemBlockMeta mMeta = blockInfo.getMeta();
-    return mMeta.isNoCache();
-  }
-
-  /**
-   * Gets the {@link UnderFileSystemBlockMeta} for a session ID and block ID pair.
-   *
-   * @param sessionId the session ID
-   * @param blockId the block ID
-   * @return the {@link UnderFileSystemBlockMeta} instance
-   * @throws BlockDoesNotExistException if the UFS block does not exist in the
-   * {@link UnderFileSystemBlockStore}
-   */
-  private BlockInfo getBlockInfo(long sessionId, long blockId) throws BlockDoesNotExistException {
-    Key key = new Key(sessionId, blockId);
-    BlockInfo blockInfo = mBlocks.get(key);
-    if (blockInfo == null) {
-      throw new BlockDoesNotExistException(ExceptionMessage.UFS_BLOCK_DOES_NOT_EXIST_FOR_SESSION,
-          blockId, sessionId);
-    }
-    return blockInfo;
-  }
-
-  /**
-   * This class is to wrap session ID amd block ID.
-   */
-  private static class Key {
-    private final long mSessionId;
-    private final long mBlockId;
+  public static final class BlockAccessToken extends UnderFileSystemBlockMeta {
```

Comment:
I might prefer to keep the Key(BlockAccessToken) as simple as we can.  make the key to extends meta looks odd to me.